### ISSUE TITLE
Interactive console: global variables (including functions) cannot be used in functions

### ISSUE THREAD

**Issue by p-l-** [secdev/scapy#1166]
How to reproduce:
```
>>> y = 1
>>> def mytest(): print(y)
---------------------------------------------------------------------------
NameError                                 Traceback (most recent call last)
<ipython-input-6-2d51dd513d71> in <module>()
----> 1 mytest()

<ipython-input-5-9edb5753faac> in mytest()
----> 1 def mytest(): print(y)

NameError: global name 'y' is not defined
```

With a function:
```
>>> def test(x): print(x)
>>> def mytest(x): test(x)
>>> mytest(12)
---------------------------------------------------------------------------
NameError                                 Traceback (most recent call last)
<ipython-input-3-7175042db3ff> in <module>()
----> 1 mytest(12)

<ipython-input-2-481b5742c96d> in mytest(x)
----> 1 def mytest(x): test(x)

NameError: global name 'test' is not defined
```

According to `git bissect`, this issue has been introduced in 887a9676436a3ed03419497b454b457ecbd13938.

**Comment by gpotter2** [comment]
This is a very well known IPython bug
https://github.com/ipython/ipython/issues/62

For some reason it exists since years...
Current only workaround is to call

`globals().update(locals())`

Before using the function

**Comment by p-l-** [comment]
I don't think we can release Scapy if it has this bug by default... @guedou?

**Comment by guedou** [comment]
It is quite a big regression =/

Not on the same topic, put I can't use completion anymore. The following does not work:
`IPv6()/ICM<TAB><TAB>`

**Comment by gpotter2** [comment]
#1169 Contains a fix

**Comment by p-l-** [comment]
Fixed by @gpotter2 in #1169. Good job!
